packages/autoloader/src/AutoloadGenerator.php

```diff
@@ -106,6 +109,117 @@ public function dump(
 		}
 	}
 
+	/**
+	 * Compiles an ordered list of namespace => path mappings
+	 *
+	 * @param  array            $packageMap  Array of array(package, installDir-relative-to-composer.json).
+	 * @param  PackageInterface $mainPackage Main package instance.
+	 *
+	 * @return array The list of path mappings.
+	 */
+	public function parseAutoloads( array $packageMap, PackageInterface $mainPackage ) {
+		$rootPackageMap = array_shift( $packageMap );
+
+		$sortedPackageMap   = $this->sortPackageMap( $packageMap );
+		$sortedPackageMap[] = $rootPackageMap;
+		array_unshift( $packageMap, $rootPackageMap );
+
+		$psr0     = $this->parseAutoloadsType( $packageMap, 'psr-0', $mainPackage );
+		$psr4     = $this->parseAutoloadsType( $packageMap, 'psr-4', $mainPackage );
+		$classmap = $this->parseAutoloadsType( array_reverse( $sortedPackageMap ), 'classmap', $mainPackage );
+		$files    = $this->parseAutoloadsType( $sortedPackageMap, 'files', $mainPackage );
+
+		krsort( $psr0 );
+		krsort( $psr4 );
+
+		return array(
+			'psr-0'    => $psr0,
+			'psr-4'    => $psr4,
+			'classmap' => $classmap,
+			'files'    => $files,
+		);
+	}
+
+	/**
+	 * Sorts packages by dependency weight
+	 *
+	 * Packages of equal weight retain the original order
+	 *
+	 * @param  array $packageMap The package map.
+	 * @return array
+	 */
+	protected function sortPackageMap( array $packageMap ) {
+		$packages = array();
+		$paths    = array();
+
+		foreach ( $packageMap as $item ) {
+			list($package, $path) = $item;
+			$name                 = $package->getName();
+			$packages[ $name ]    = $package;
+			$paths[ $name ]       = $path;
```

Comment:
I'm surprised WordPress's linting rules don't complain about this not having spaces inside the parens.
```suggestion
			list( $package, $path ) = $item;
			$name                   = $package->getName();
			$packages[ $name ]      = $package;
			$paths[ $name ]         = $path;
```